### ISSUE TITLE
Empty Repositories after Update to latest 13.4.1

### ISSUE THREAD

**Issue by RolandSM** [docker/for-linux#1113]
Hello,

I run a docker-ce container on my debian server and today I made an update from 13.3.6 to 13.4.1 with the Portainer recreate, pull latest as i do it since years.
My docker config, data, logs folders are persistent from host.

The update worked and after a short check, all projects are available, but if I got into a project Gitlab shows an empty repository.
I checked the data folder git-data/repositories/ all project folders are here, also sub-project folders. But inside those nothing.

My data folder is over 4 GB so I think my chances are not to bad to get my repositories back???
I removed 13.4.1 and downloaded again 13.3.6 and started the container but same, empty repositories.

I am so stupid, most on updates I copied my data folder, but the last updates like named was without problems, so today I have no other backup.

Is there a internal backup mechanism and I can restore the yesterday system where everything exists?

**Comment by RolandSM** [comment]
ah seems not every repostory is gone away???
Of course I have all locally in my IDEs.

**Comment by RolandSM** [comment]
After further searching I found the Legacy-Hashed-Projects Topic like:
[https://docs.gitlab.com/ee/administration/raketasks/storage.html#migrate-existing-projects-to-hashed-storage](url)

The script can find 89 projects using legacy storage.
If I try to migrate and monitor all attempts failed.

**Comment by RolandSM** [comment]
Ouh. hours later I was able to fix all my problems.
The topic was the legacy storage, I had 89 projects.
My first attempts failed with OpenSSL-Sipher-Error.

Then I removed all runnger registration tokens
[https://docs.gitlab.com/ee/raketasks/backup_restore.html#when-the-secrets-file-is-lost](url)

Afterwards the doctor:secrets showed only a few errors but no project token errors.
All of my 89 projects sayed runners_token failures before...

After this I made a new migrate_to_hased and then in monitoring without errors.

Now the storage:legacy command shows 0 and my projects works again.
